appGeo/src/tests/game_tests.spec.ts

```diff
@@ -0,0 +1,231 @@
+
+    assert.equal(game1.removeChat(chat3.getChatID()), chat3, "Can successfully remove Chat");
+    chatArray.pop();
+    assert.deepEqual(game1.getChats(), chatArray, "chat list successfully updated");
+
+    assert.equal(game1.getChat(chat3.getChatID()), "", "A lack of Chat should return failure");
+
+
+});
+
+QUnit.test("Game Start and End", function(assert) {
+    const now = new Date();
+    const next_week = new Date(now.getTime() + 7 * 24 * 60 * 60 * 1000);
+
+    const test_map = new CampusMap();
+
+    const Location1 = new Location(1);
+
+    const player1 = new Player(1, 'player1', Location1, ALIVE); 
+    const player2 = new Player(1, 'player2', Location1, ALIVE);
+    const player3 = new Player(1, 'player3', Location1, ALIVE);
+
+   const playerArray = [player1, player2, player3];
+
+   const playersMap = new Map()
+    for (const p of playerArray) {
+        playersMap.set(p.userID, p)
+      }
+
+    const test_game = new Game(next_week, test_map, playersMap);
+
+    assert.equal(test_game.startGame(), FAILURE, "Game cannot start with too few players");
+    assert.false(test_game.isGameActive(), "Unstarted game is not active");
+    assert.equal(test_game.endGame(),FAILURE, "Unstarted game cannot end");
+    assert.false(test_game.isGameActive(), "Unstarted game is not active");
+
+    const player4 = new Player(1, 'player4', Location1, ALIVE);
+    const player5 = new Player(1, 'player5', Location1, ALIVE);
+
+    test_game.addPlayer(player4);
+    test_game.addPlayer(player5);
+
+    assert.equal(test_game.PlayerCount, 3, "Can count players outside of game");
+    assert.equal(test_game.getRoleCount(CIVILIAN), 0, "No role assigned, no civilians");
+    assert.equal(test_game.getRoleCount(KILLER), 0, "No role assigned, no killers");
+
+    assert.equal(test_game.getFractionRole(CIVILIAN), 0, "No role assigned, no civilians");
+    assert.equal(test_game.getFractionRole(KILLER), 0, "No role assigned, no killers");
+
+    assert.equal(test_game.endGame(),FAILURE, "Unstarted game cannot end");
+    assert.false(test_game.isGameActive(), "Unstarted game is not active");
+
+    assert.equal(test_game.startGame(), SUCCESS, "Game can start with five or more players");
+    assert.true(test_game.isGameActive(), "Started game is active");
+
+    assert.equal(test_game.PlayerCount, 5, "Can count players in game");
+    assert.equal(test_game.getRoleCount(CIVILIAN), 3, "Should start game with 3 civilians");
+    assert.equal(test_game.getRoleCount(KILLER), 1, "Should start game with 1 killer");
+
+    assert.equal(test_game.getFractionRole(CIVILIAN), 0.8, "Should be 80% civilians");
+    assert.equal(test_game.getFractionRole(KILLER), 0.2, "Should be 20% killers");
+
+    assert.equal(test_game.endGame(), SUCCESS, "Can stop a game");
+    assert.false(test_game.isGameActive(), "Stopped game is not active");
+
+    playersMap.set(player4.getUserID, player4);
+    playersMap.set(player5.getUserID, player5);
```

Comment:
Should be called as player4.getUserID() [Forgot the parenthesis so not a function call]